src/hyperion/external_interaction/ispyb/store_datacollection_in_ispyb.py

```diff
@@ -286,19 +303,37 @@ def _mutate_data_collection_params_for_experiment(
         return params
 
     def _store_scan_data(self, conn: Connector):
-        if not self.data_collection_group_id:
-            self.data_collection_group_id = self._store_data_collection_group_table(
-                conn
-            )
-        data_collection_id = self._store_data_collection_table(
-            conn, self.data_collection_group_id
-        )
-        self.data_collection_id = data_collection_id
-        self._store_position_table(conn, data_collection_id)
+        assert (
+            self.data_collection_group_id
+        ), "Attempted to store scan data without a collection group"
+        assert (
+            self.data_collection_id
+        ), "Attempted to store scan data without a collection"
+        self._store_data_collection_group_table(conn)
+        self._store_data_collection_table(conn, self.data_collection_group_id)
+        self._store_position_table(conn, self.data_collection_id)
 
-        return data_collection_id, self.data_collection_group_id
+        return self.data_collection_id, self.data_collection_group_id
 
     def begin_deposition(self) -> IspybIds:
```

Comment:
Since we always want to do this at the run document, it might be worth calling it in `ispyb_callback_base`'s `activity_gated_start`